modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro

```diff
@@ -1048,6 +1062,46 @@ definition {
 		Button.clickSave();
 	}
 
+	macro editObjectFolderViaUi {
+		Click(locator1 = "ObjectAdmin#FOLDER_ACTIONS");
+
+		if (${optionSetValue} == "Edit Value and ERC") {
+			Click(locator1 = "ObjectAdmin#EDIT_LABEL_AND_ERC");
+		}
+
+		if (isSet(newLanguage)) {
+			ObjectAdmin.changeLabelLanguage(newLanguage = ${newLanguage});
+		}
+
+		if (isSet(labelName)) {
+			CreateObject.editLabel(labelName = ${labelName});
+		}
+
+		if (isSet(externalReferenceCode)) {
+			Type(
+				key_externalReferenceCode = ${externalReferenceCode},
+				locator1 = "ObjectAdmin#EXTERNAL_REFERENCE_CODE_FOLDER");
+		}
+
+		if (${optionSetValue} == "Folder Permissions") {
+			Click(locator1 = "ObjectAdmin#EDIT_FOLDER_PERMISSIONS");
+		}
+
+		if (${optionSetValue} == "Delete Folder") {
+			Click(locator1 = "ObjectAdmin#DELETE_FOLDER");
+
+			SelectFrame(locator1 = "ObjectCustomViews#VIEW_MODAL_CONTENT");
+
+			Type(
+				locator1 = "ObjectAdmin#PLACEHOLDER_DELETE_OBJECT",
+				value1 = ${objectName});
+
+			Click(locator1 = "Form#SIDEBAR_DELETE_FIELD_MODAL_DELETE_BUTTON");
+		}
```

Comment:
@thaiscabrals I like the ifs under` Edit Label and ERC` because I like how it gives you the option to edit a field or not. 

But what do you think about creating separate macros for each menu item (e.g. editFolderPermissions, deleteFolder)? I think doing so will make it easier to read and maintain even in the test case level.